api/serializers/product_serializer.py

```diff
@@ -0,0 +1,51 @@
+from retail_app.models import Product
+from . import (
+    PriceSerializer,
+    ImageSerializer,
+    StockSerializer,
+    KeywordsSerializer,
+)
+import json
+
+
+class ProductSerializer:
+    def __init__(self, product: Product):
+        self.product = product
+
+    def product_stock(self):
+        return self.product.productstock_set.all()
+
+    def product_images(self):
+        return self.product.productimage_set.all()
+
+    def product_keywords(self):
+        return self.product.searchproductkeywords_set.all()
```

Comment:
I wouldn't include these in your results. This is more for internal classification. By adding this to the api you are publicly exposing it and apps may begin to rely on it.

Reply:
this should be added to the product class?

Reply:
I would take it out altogether.

Reply:
but don't i need to set the foreign keys before i can use them?

Reply:
I am referring to the serializer only, the models should be left as they are. The serializer should not expose the keywords as part of the serialization. The serializer gives you the ability to choose what you want to expose and what you want to keep hidden, like public/private methods, the api should only expose the public pieces. The keywords model is used as a backing mechanism for search logic, it should not expose that to the outside world in the api.

Reply:
😅